python/cuml/tests/test_linear_svm.py

```diff
@@ -253,23 +259,116 @@ def run_classification(datatype, penalty, loss, dims, nclasses):
     X_test = X_test.get()
     y_train = y_train.get()
     y_test = y_test.get()
+    cud = cud.get()
     gc.collect()
 
     try:
 
         def run_sklearn():
             skm = sk.LinearSVC(
-                loss=loss, penalty=penalty, max_iter=skit, dual=skdual
+                loss=loss,
+                penalty=penalty,
+                max_iter=skit,
+                dual=skdual,
+                class_weight=class_weight,
             )
             skm.fit(X_train, y_train)
-            return skm.score(X_test, y_test)
+            return skm.score(X_test, y_test), skm.decision_function(X_test)
 
-        sks = with_timeout(timeout=t, target=run_sklearn)
+        sks, skd = with_timeout(timeout=t, target=run_sklearn)
         good_enough(cus, sks, nrows)
+
+        # always confirm correct shape of decision function
+        assert cud.shape == skd.shape, (
+            f"The decision_function returned different shape "
+            f"cud.shape = {cud.shape}; skd.shape = {skd.shape}))"
+        )
+
     except TimeoutError:
         pytest.skip(f"sklearn did not finish within {t} seconds.")
 
 
+@pytest.mark.parametrize("datatype", [np.float32, np.float64])
+@pytest.mark.parametrize("penalty", ["l2"])
+@pytest.mark.parametrize("loss", ["squared_hinge"])
+@pytest.mark.parametrize(
+    "dims",
+    [
+        unit_param((3, 1)),
+        unit_param((1000, 10)),
+        unit_param((100, 100)),
+    ],
+)
+@pytest.mark.parametrize("nclasses", [2, 3, 5, 8])
```

Comment:
what's the reason behind these numbers of classes? Maybe `2, 16` might be enough and provide more coverage?